File: expense-tracker/expense-tracker/src/main/java/com/shivam/expense_tracker/service/DashboardService.java
package com.shivam.expense_tracker.service;

import java.math.BigDecimal;
import java.util.List;

import com.shivam.expense_tracker.dto.dashboard.CategoryChartResponse;
import com.shivam.expense_tracker.dto.dashboard.DashboardResponse;
import com.shivam.expense_tracker.dto.dashboard.MonthlyTrendResponse;

public interface DashboardService {

    DashboardResponse getDashboardSummary(Long userId, Integer month, Integer year);

    DashboardResponse getDashboard(Long userId);

    List<CategoryChartResponse> getCategoryWiseExpense(Long userId, Integer month, Integer year);

    List<MonthlyTrendResponse> getMonthlyTrend(Long userId, Integer year);

    BigDecimal getMonthlyExpense(Long userId, Integer month, Integer year);
    
    BigDecimal getTotalExpenseByMonth(Long userId, Integer month, Integer year);
    

	DashboardResponse getDashboard();
   
}

File: expense-tracker-frontend/src/services/dashboardService.js
import api from "./api";

export const getDashboardSummary = (userId, month, year) =>
    api.get(`/dashboard/${userId}`, {
        params: { month, year }
    });

export const getSpendingByCategory = (userId) =>
  api.get(`/dashboard/category/${userId}`);

export const getMonthlyTrend = (userId, year) =>
  api.get(`/dashboard/monthly/${userId}`,{
    params: {year}
  });

export const getDashboard = (userId) =>
  api.get(`/dashboard/${userId}`);

export const getCategoryWiseExpense = (userId, month, year) =>
    api.get(`/dashboard/category/${userId}`, {
        params: { month, year }
    });

export const getMonthlyExpense = (userId, month, year) =>
  api.get(`/dashboard/monthly-expense/${userId}`, {
    params: { month, year },
  });